### DISCUSSION TITLE
How to upgrade on Apple Silicon

### DISCUSSION THREAD

**Discussion by raffetazarius** [vladmandic/sdnext#3115] [Q&A]
I'm following instructions here https://github.com/vladmandic/automatic/wiki/Advanced-Install#upgrades to upgrade SD Next

Neither the --upgrade command arg, nor doing a git pull gets me updated from the 2023-03-21 version.

I have modified the /models folder in my installation with a symlink to a shared model folder, but other than that it's a "factory" install.

[--upgrade.txt](https://github.com/vladmandic/automatic/files/15278642/--upgrade.txt)
[git pull.txt](https://github.com/vladmandic/automatic/files/15278643/git.pull.txt)

---

Computer: MacBook Pro 2024 M3 Max 128GB Memory 8TB SSD

Starting SD.Next                                                                                                                        
11:10:39-465587 INFO     Logger: file="/Users/privateuser/automatic/sdnext.log" level=INFO size=207252 mode=append                                             
11:10:39-466158 INFO     Python 3.10.13 on Darwin                                                                                                                
11:10:39-485766 INFO     Version: app=sd.next updated=2024-03-21 hash=82973c49 branch=master url=https://github.com/vladmandic/automatic/tree/master             
11:10:39-931665 INFO     Updating main repository                                                                                                                
11:10:40-004407 ERROR    Error running git: None / stash                                                                                                         
11:10:40-413225 ERROR    Error running git: . / pull --rebase --force                                                                                            
11:10:40-422760 INFO     Upgraded to version: 82973c49 Thu Mar 21 11:23:50 2024 +0900                                                                            
11:10:40-426493 INFO     Platform: arch=arm64 cpu=arm system=Darwin release=23.4.0 python=3.10.13       

full startup log is attached.

**Comment by vladmandic** [comment]
that is the latest RELEASE.
if you want newer code, you need to switch to DEV branch - search for that.
new release is planned soon, likely next week.

**Reply by raffetazarius** [reply]
ah i see ok, so none of those errors my log point to --upgrade NOT working when there's a new release out?

**Reply by vladmandic** [reply]
well, you did modify some not-allowed parts of `models`. if you modify any file/folder which is under git control, git will refuse to update.
so right now there is nothing to update to as you're on latest release anyhow.
but once new release is out, you wont be able to easily update.

```log
11:10:40-442543 WARNING  Deleted files: ['models'.....<redacted>]                                                                                                       
11:10:40-445677 WARNING  Modified files: ['models/...<redacted>...']                                                                                                       
```

and yes, you can use `git pull`, but that only updates core repo, it doesn't update submodules.
that's why its recommended to use `--upgrade` that handles everything.
